src/intra_edge.rs

```diff
@@ -103,58 +158,57 @@ impl EdgeBranch {
     const fn new(edge_flags: EdgeFlags, bl: BlockLevel) -> Self {
         let o = edge_flags;
         let h = [
-            edge_flags | EDGE_LEFT_HAS_BOTTOM,
-            edge_flags & EDGE_LEFT_HAS_BOTTOM,
+            edge_flags.union(EdgeFlags::LEFT_HAS_BOTTOM),
+            edge_flags.intersection(EdgeFlags::LEFT_HAS_BOTTOM),
         ];
         let v = [
-            edge_flags | EDGE_TOP_HAS_RIGHT,
-            edge_flags & EDGE_TOP_HAS_RIGHT,
+            edge_flags.union(EdgeFlags::TOP_HAS_RIGHT),
+            edge_flags.intersection(EdgeFlags::TOP_HAS_RIGHT),
         ];
         let node = EdgeNode { o, h, v };
 
         let h4 = [
-            edge_flags | EDGE_LEFT_HAS_BOTTOM,
-            EDGE_LEFT_HAS_BOTTOM
-                | (if bl == BL_16X16 {
-                    edge_flags & EDGE_I420_TOP_HAS_RIGHT
-                } else {
-                    0 as EdgeFlags
-                }),
-            EDGE_LEFT_HAS_BOTTOM,
-            edge_flags & EDGE_LEFT_HAS_BOTTOM,
+            edge_flags.union(EdgeFlags::LEFT_HAS_BOTTOM),
+            EdgeFlags::LEFT_HAS_BOTTOM.union(
+                (edge_flags.intersection(EdgeFlags::I420_TOP_HAS_RIGHT)).select(bl == BL_16X16),
```

Comment:
```suggestion
                edge_flags.intersection(EdgeFlags::I420_TOP_HAS_RIGHT).select(bl == BL_16X16),
```

Reply:
that makes this format to 3 lines instead of 1, I'm happy to go with that but the parens did have a reason